### PR TITLE
Add LZMA support to CMake build

### PR DESCRIPTION
**Pull request by JosiahWI**
This looks for the LZMA library and sets HAVE_LZMA_H if found. It also links the inkcache library to LZMA.

### CODE REVIEW

**Comment by cmcfarlen** [comment]
[approve ci autest]

**Comment by JosiahWI** [comment]
[approve ci autest]

File: include/tscore/ink_config.h.cmake.in [hunk 1]
```diff
@@ -104,6 +104,7 @@ const int DEFAULT_STACKSIZE = @DEFAULT_STACK_SIZE@;
 #define TS_MAX_HOST_NAME_LEN @TS_MAX_HOST_NAME_LEN@
 
 /* Feature Flags */
+#cmakedefine01 HAVE_LZMA_H
```
Comment (cmcfarlen) [inline include/tscore/ink_config.h.cmake.in:107]: Move and use `#cmakedefine`

**Comment by JosiahWI** [comment]
[approve ci autest]

**Comment by JosiahWI** [comment]
[ci approve autest]

**Comment by JosiahWI** [comment]
[approve ci autest]

**Comment by JosiahWI** [comment]
[approve ci autest]

**Pull request merged**